src/ResourceManagement/ResourceManager/Authentication/AzureCredentials.cs

```diff
@@ -80,6 +88,12 @@ public AzureCredentials WithDefaultSubscription(string subscriptionId)
 
         public async override Task ProcessHttpRequestAsync(HttpRequestMessage request, CancellationToken cancellationToken)
         {
+            if (this.msiTokenProvider != null)
+            {
+                request.Headers.Authorization = await this.msiTokenProvider.GetAuthenticationHeaderAsync(cancellationToken);
```

Comment:
Credentials won't be cached explicitly. The token provider (MSI extension) running on VM will be responsible for caching the auth token.